ams/AssetAnalyzer.cs

```diff
@@ -96,73 +125,137 @@ await foreach (var locator in locators)
+                statistics = new AssetStats();
+                assetTypes = new ConcurrentDictionary<string, int>();
 
-            if (resourceFilter != null)
-            {
-                // When a filter is used, it usually include a small list of assets,
-                // The total count of asset can be extracted in advance without much perf hit.
-                filteredList = await assets.ToListAsync(cancellationToken);
+                List<MediaAssetResource>? filteredList = null;
 
-                totalAssets = filteredList.Count;
-            }
+                if (resourceFilter != null)
+                {
+                    // When a filter is used, it usually include a small list of assets,
+                    // The total count of asset can be extracted in advance without much perf hit.
+                    filteredList = await assets.ToListAsync(cancellationToken);
+
+                    totalAssets = filteredList.Count;
+                }
 
-            _logger.LogInformation("The total assets to handle in this run is {count}.", totalAssets);
+                _logger.LogInformation("The total assets to handle in this run is {count}.", totalAssets);
 
-            var channel = Channel.CreateBounded<double>(1);
-            var progress = ShowProgressAsync("Analyzing Assets", "Assets", totalAssets, channel.Reader, cancellationToken);
-            var writer = channel.Writer;
-            await MigrateInParallel(assets, filteredList, async (asset, cancellationToken) =>
+                var channel = Channel.CreateBounded<double>(1);
+                var progress = ShowProgressAsync("Analyzing Assets", "Assets", totalAssets, channel.Reader, cancellationToken);
+                var writer = channel.Writer;
+                await MigrateInParallel(assets, filteredList, async (asset, cancellationToken) =>
+                {
+                    var storage = await _resourceProvider.GetStorageAccountAsync(account, asset, cancellationToken);
+                    var result = await AnalyzeAsync(asset, storage, cancellationToken);
+                    var assetType = result.AssetType ?? "unknown";
+                    assetTypes.AddOrUpdate(assetType, 1, (key, value) => Interlocked.Increment(ref value));
+                    reportGenerator.WriteRecord(result);
+                    statistics.Update(result);
+                    await writer.WriteAsync(statistics.Total, cancellationToken);
+                },
+                _analysisOptions.BatchSize,
+                cancellationToken);
+
+                writer.Complete();
+                await progress;
+                _logger.LogDebug("Finished analysis of assets for account: {name}. Time taken {elapsed}", _analysisOptions.AccountName, watch.Elapsed);
+            }
+            catch (Exception ex)
+            {
+                if (ex.Message.Contains("was not found"))//ex.Data is SomeExceptionType someException && someException.ErrorCode == "ResourceNotFound")
```

Comment:
Does it throw a specific type of exception when the account is not found?  it seems not a common way to check the message with "was not found".

Reply:
To check if the input parameter is for an AMS account, the code should just use try ... catch for                 var account = await GetMediaAccountAsync(_analysisOptions.AccountName, cancellationToken); only,  If the valid account is returned, it is AMS account, otherwise treat it as Storage Account.

With current code, the exception could still be thrown if it is a valid AMS account, but some other condition is not met.